drivers/Makefile.dep

```diff
@@ -204,6 +204,13 @@ ifneq (,$(filter ccs811,$(USEMODULE)))
   USEMODULE += xtimer
 endif
 
+ifneq (,$(filter cst816s,$(USEMODULE)))
+  FEATURES_REQUIRED += periph_gpio_irq
+  FEATURES_REQUIRED += periph_gpio
+  FEATURES_REQUIRED += periph_i2c
+  USEMODULE += xtimer
+endif
+
```

Comment:
Now each driver add its own Makefile.dep. Here it would have the following content:

```mk
FEATURES_REQUIRED += periph_gpio_irq
FEATURES_REQUIRED += periph_i2c
USEMODULE += xtimer
```

(no need to pull periph_gpio since it's pulled automatically with periph_gpio_irq:
https://github.com/RIOT-OS/RIOT/blob/185f703c23471f6e6da84325528ea36ea337cacd/Makefile.dep#L1067-L1070
)

You can also remove the changes in Makefile.dep:
```suggestion
```